packages/recommend-react/src/useRecommendations.ts

```diff
@@ -19,35 +20,51 @@ export function useRecommendations<TObject>({
   recommendClient,
   threshold,
   transformItems: userTransformItems,
-}: GetRecommendationsProps<TObject>) {
-  const [result, setResult] = useState<GetRecommendationsResult<TObject>>({
-    recommendations: [],
-  });
+  initialState: userInitialState,
+}: GetRecommendationsProps<TObject> & {
+  initialState?: InitialState<TObject>;
+}) {
+  const isFirstRenderRef = useRef(true);
+
   const { status, setStatus } = useStatus('loading');
   const objectIDs = useStableValue(userObjectIDs);
   const transformItems = useStableValue(userTransformItems);
   const queryParameters = useStableValue(userQueryParameters);
   const fallbackParameters = useStableValue(userFallbackParameters);
 
+  const initialState = useStableValue<GetRecommendationsResult<TObject>>({
+    recommendations: [],
+    ...userInitialState,
+  });
+  const [recommendationsResult, setRecommendationsResult] = useState<
```

Comment:
We can rollback to the previous naming with `[result, setResult]`.

Reply:
I was thinking that the naming now is a bit more descriptive for what it represents 🤔

Reply:
In this context we know that we're talking about recommendations.

Reply:
✅